### PR TITLE
Implement AzCopy runner

### PR DESCRIPTION
**Pull request by adreed-msft**
This is a _very_ basic implementation of things, and _very_ unfinished. It is more intended to ship a semi-functional prototype of the new E2E framework that real-world feedback can be generated against than to bring it to production readiness.

### CODE REVIEW

File: e2etest/newe2e_resource_manager_azstorage.go [hunk 1]
```diff
@@ -17,6 +22,112 @@ type AzureAccountResourceManager struct {
...
 }
 
+func (acct *AzureAccountResourceManager) ApplySAS(a Asserter, URI string, loc common.Location) string {
+	a.AssertNow("account must not be nil", Not{IsNil{}}, acct)
+	switch loc {
+	case common.ELocation.Blob():
+		skc, err := blob.NewSharedKeyCredential(acct.accountName, acct.accountKey)
+		a.NoError("Create shared key", err)
+
+		p, err := blobsas.AccountSignatureValues{ // todo: currently this generates a "god SAS", granting all permissions. This is bad! Let's not do this in prod.
```
Comment (gapra-msft) [inline e2etest/newe2e_resource_manager_azstorage.go:32]: a thought for the future, maybe we make this configurable for the user to pass in a string that represents the sas permissions they want (ideally the user passes the minimal required) and we can parse it as a SignatureValues obj and pass it here

File: e2etest/newe2e_resource_managers_blob.go [hunk 2]
```diff
@@ -83,8 +83,14 @@ func (b *BlobServiceResourceManager) ListContainers(a Asserter) []string {
 	return out
 }
 
-func (b *BlobServiceResourceManager) URI() string {
-	return blobStripSAS(b.internalClient.URL())
+func (b *BlobServiceResourceManager) URI(a Asserter, withSas bool) string {
+	base := blobStripSAS(b.internalClient.URL())
+
+	if withSas {
```
Comment (gapra-msft) [inline e2etest/newe2e_resource_managers_blob.go:89]: why do we need to apply sas below? does the internalClient.URL not contain SAS?

**Review by gapra-msft — APPROVED** [review]

File: e2etest/newe2e_resource_managers_blob.go (hunk 2 above)
Comment (adreed-msft) [reply e2etest/newe2e_resource_managers_blob.go:89]: Nope. It's using account key based validation.

File: e2etest/newe2e_resource_manager_azstorage.go (hunk 1 above)
Comment (adreed-msft) [reply e2etest/newe2e_resource_manager_azstorage.go:32]: Great idea! I was planning on addressing this alongside some stronger auth method selection for RunAzCopy.

**Review by tasherif-msft — APPROVED** [review]

**Pull request closed without merging**